core/lib/db_connection/src/metrics.rs

```diff
@@ -0,0 +1,14 @@
+use std::time::Duration;
+
+use vise::{Buckets, Histogram, LabeledFamily, Metrics, Unit};
+
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "sql_connection")]
+pub(crate) struct ConnectionMetrics {
```

Comment:
I'd expect that *all* metrics defined in the DAL crate would be moved here, together with the `instrument` module and macros (the macros would need to be exported in order to work). Are there arguments against sharing these among DAL crates?